src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  let anotherState = { ...state };
+  const massState = [];
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    if (type === 'addProperties') {
+      Object.assign(anotherState, extraData);
+    }
+
+    if (type === 'removeProperties') {
+      for (const del of keysToRemove) {
+        delete anotherState[del];
+      }
+    }
 
+    if (type === 'clear') {
+      anotherState = {};
+    }
+
+    massState.push({ ...anotherState });
```

Comment:
The action of creating a copy of 'anotherState' and pushing it into 'massState' is repeated in every condition. To adhere to the DRY principle, this action should be performed once after the 'if' or 'switch' statement.